### PR TITLE
Use chart to show TagStat 

### PR DESCRIPTION
**Pull request by jybao1**
### Changes:
- Added a google chart module and ``donut-chart`` component to use google chart library. 
- Added ``fetchTagStats`` in ``DataService`` to fetch TagStats from backend service.
- Modified ``thread-detail.component.html``. When tagStats is available and non-empty, a donut chart will be rendered. When no tagStats is available, a notification will show. 

### Screenshot:
- Show a donut chart when tagStats is available.
screencapture-localhost-4200-thread-a159fe63-6b40-4e3d-9978-c0af9bfab1cb-tagName-testTag-2020-08-04-01_12_11

- Show a notification if no tagStats found for the given tag.
screencapture-localhost-4200-thread-0ad0ff0c-9e4b-492c-a291-5cb7cd786fd2-tagName-noise-2020-08-04-01_11_45

### CODE REVIEW

File: ui/src/app/google-chart/donut-chart/donut-chart.component.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    this.library = this.chartService.getGoogle();
+    this.library.charts.load('current', { packages: ['corechart'] });
+    this.library.charts.setOnLoadCallback(this.drawChart.bind(this));
+  }
+
+  ngOnInit(): void {
+  }
+
+  private drawChart(): void {
+    const chart = new this.library.visualization.PieChart(document.getElementById('donutchart'));
```
Comment (XuZehua-Teapot) [inline ui/src/app/google-chart/donut-chart/donut-chart.component.ts:30]: Consider using a template reference variable and inject it with [`ViewChild`](https://angular.io/api/core/ViewChild).

File: ui/src/app/google-chart/donut-chart/donut-chart.component.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ */
+export class DonutChartComponent implements OnInit {
+
+  private library;
+  @Input() tagStats: TagStats; // Input passed from ThreadDetailComponent
+
+  constructor(private chartService: GoogleChartService) {
+    this.library = this.chartService.getGoogle();
+    this.library.charts.load('current', { packages: ['corechart'] });
+    this.library.charts.setOnLoadCallback(this.drawChart.bind(this));
```
Comment (XuZehua-Teapot) [inline ui/src/app/google-chart/donut-chart/donut-chart.component.ts:23]: The callback can be set in the `load()` call.

File: ui/src/app/google-chart/donut-chart/donut-chart.component.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ * A donut chart component to visualize TagStats
+ */
+export class DonutChartComponent implements OnInit {
+
+  private library;
+  @Input() tagStats: TagStats; // Input passed from ThreadDetailComponent
+
+  constructor(private chartService: GoogleChartService) {
+    this.library = this.chartService.getGoogle();
+    this.library.charts.load('current', { packages: ['corechart'] });
```
Comment (XuZehua-Teapot) [inline ui/src/app/google-chart/donut-chart/donut-chart.component.ts:22]: Ideally this should be loaded only once.

Not sure what's the best practice in Angular to load an asynchronous resource. Maybe wrap it in a `Promise` and return it from `GoogleChartService`?

File: ui/src/app/google-chart/google-chart.service.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  private readonly google: any;
+  private loaderPromise: Promise<any>;
+  number = 1;
+
+  constructor() {
+    this.google = google;
+  }
+
+  getGoogle(): any {
```
Comment (XuZehua-Teapot) [inline ui/src/app/google-chart/google-chart.service.ts:21]: Is this still needed?

File: ui/src/app/google-chart/google-chart.service.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ * Service to access Google Chart library
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class GoogleChartService {
+
+  private readonly google: any;
+  private loaderPromise: Promise<any>;
+  number = 1;
```
Comment (XuZehua-Teapot) [inline ui/src/app/google-chart/google-chart.service.ts:15]: What's this?

File: ui/src/app/google-chart/google-chart.service.ts (hunk 4 above)
Comment (jybao1) [reply ui/src/app/google-chart/google-chart.service.ts:20]: I think it is still needed for library access in ``DonutChartComponent`` ?

File: ui/src/app/google-chart/google-chart.service.ts (hunk 5 above)
Comment (jybao1) [reply ui/src/app/google-chart/google-chart.service.ts]: Removed.

**Pull request merged**